### PR TITLE
Fix not cleared storage task handlers from setImmediate

### PR DESCRIPTION
**Pull request by mazulatas**
**Description:**
This PR adds cleaning up handle tasks in Immediate

### CODE REVIEW

**Comment by stylesam** [comment]
@cartant Why did the build fall?

**Comment by cartant** [comment]
> Why did the build fall?

It's nothing you've done The version of `@types/node` is too old for TypeScript 3.6.3. That can be dealt with elsewhere.

I'll review your PR a little later. It looks good; there's just one or two things I'd like changed. Thanks.

**Comment by stylesam** [comment]
@cartant Thanks for review, we are waiting you

File: src/internal/util/Immediate.ts [hunk 1]
```diff
@@ -21,3 +22,9 @@ export const Immediate = {
     delete tasksByHandle[handle];
   },
 };
+
+export const ForTests = {
```
Comment (cartant) [inline src/internal/util/Immediate.ts]: I'd suggest renaming `ForTests` to `TestTools` and renaming `numberOfTasksToHandle` to `pending`.

Comment (mazulatas) [reply src/internal/util/Immediate.ts]: I agree

**Comment by benlesh** [comment]
Thank you @mazulatas for catching this one and bringing it to my attention. I've further refactored the code to make it a little more efficient than it was, and I've kept your test and test util.

(I did the code editing in the browser, so we'll see if CI passes, haha)

**Pull request merged**